architecture/overview.md

```diff
@@ -0,0 +1,66 @@
+
+### Concepts
+
+CockroachDB heavily relies on the following concepts, so understanding them will help you understand what our architecture achieves.
+
+Term | Definition
+-----|-----------
+**Consistency** | CockroachDB uses this term in a few different ways, but most commonly as consistency in terms of the [CAP theorem](https://en.wikipedia.org/wiki/CAP_theorem). In this context, consistency means that every read receives the most recent write or an error. Most often we refer to consistency as the CAP theorem guarantee we prioritize in face of all others, making us what's known as "CP" database.<br/><br/>However, CockroachDB does also also support [consistency in terms of ACID semantics](https://en.wikipedia.org/wiki/Consistency_(database_systems)). To confuse matters slightly, what we typically refer to as consistency (in the CAP theorem sense), is also tightly coupled to our support of [isolation in terms of ACID semantics](https://en.wikipedia.org/wiki/Isolation_(database_systems)).
+**Consensus** | When an action is performed on a range, a quorum of nodes containing a replica of it acknowledge the action. When an action *doesn't* achieve consensus, forward progress halts to ensure consistency within the cluster.
+**Replication **| Copying ranges from one node to another, or applying changes made on one replica of a range to others. Each node containing a replica communicates with all other nodes that contain that data to ensure any changes to it achieve consensus.
+**Transactions** | A set of operations performed on your database that satisfy the requirements of [ACID semantics](https://en.wikipedia.org/wiki/Database_transaction). This is a crucial component for a consistent system to ensure developers can trust the data in their database.
+**Multi-Active Availability** | A consensus-based notion of high availability that lets all nodes in a cluster receive reads and writes, even during a failover, without creating any conflicts. This is similar to active-active replication, but circumvents the possibility of nodes having conflicting information.
+
+## Overview
+
+Developers interact with CockroachDB through a SQL API, which we've modeled after PostgreSQL. These requests can be sent to any node; all of them behave symmetrically.
+
+All that's required to add a node to the cluster is starting the `cockroach` process with a `join` flag set to the host of any other node in the cluster.
+
+After receiving a SQL RPC, the node converts the statement into operations that work with our distributed key-value store. These operations have access to a cache which details where each piece of data is laid out on your cluster, where they go and perform reads and writes.
+
+Data is distributed algorithmically by CockroachDB, with all data being replicated >=3 times to ensure survivability. This way, if nodes go down, you still have copies of the data which can be used for reads and writes, as well as replicating the data to any new nodes that join.
```

Comment:
We should spend another sentence making it clear that the number of replicas may not equal the number of nodes. This seems to be one of the most common misunderstandings about our architecture.